### DISCUSSION TITLE
Help getting started using GitHub pls

### DISCUSSION THREAD

**Discussion by JoahfromAlberta** [community/community#72974] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Can you provide some direction to help replicate this app to work on our shopify website.  So far we have created an account on github we have installed the git hub desktop and cloned the repository on our local machine.  https://github.com/JoahfromAlberta/yugioh-deck-tool

The author suggests in development: 2) Execute npm ci inside the cloned directory. What is the purpose of this? And do we require to open a shell to do this or can we do this from desktop. 

Any help would be kindly appreciated,

**Comment by rockreyad** [comment]
Hello @JoahfromAlberta ,
the project you shared , this is a [vuejs](https://vuejs.org/guide/quick-start.html#creating-a-vue-application) project which created with [vite](https://vitejs.dev/guide/) .

and `npm ci` stands for clean install all of your `node_modules` those are used during development of these project. you can find those installed modules list in `package.json` file. actually the  `node_modules` contains all of your dependency.  This allows the developer to focus on the business logic and create the application more quickly. So when you hit these `npm ci` command all the external dependencies downloading to `node_modules` folder.

and the `npm` command only work on your shell if [nodejs](https://nodejs.org/en) is installed on your machine.

regrading , replicate this app to your Shopify store may need [Shopify theme customization](https://help.shopify.com/en/manual/online-store/themes/customizing-themes) or Apps Development to achieve these kind of work .

I trust this information helps you,

Thank You

**Reply by JoahfromAlberta** [reply]
Hi Rockreyard, thank you so much for your reply.  I did have a look at some excellent youtube videos and understand so far the need to use a terminal to run github for greater understanding of the system hierarchy and the mandatory use of SSH key.  We have a similar key to use for our protection when streaming.  My greatest challenge is to keep things in moderation and not go too fast as to become overwhelmed :) 

So npm ci will pull the modules to install for development.  I imagine that if it can't find these modules it will give errors otherwise it should be ok.  It must reference a build file in the repository to see what modules are required.

We did get Shopify to recognize our repositiory yugiohdeckpro on our github but then it returned an error and when we tried agian it just offered a blank.  Mayb permissions?  that we are not too overly concerned with atm, we still need to see this thing work before proceeding.

In all honesty, 1) we only need one value edit in the working app on yugiohprodeck, the developer was nice enough to highlight the line that requires the chance on their repository.  We have used the recommended editor to update the value on our cloned repository but when we hit the html.index it does not run the app it only shows the first few lines of text in our web browser waaa we asked him if he could please help but he cited he does not want to help due to time constraints.
 2) we do wish to learn the proper way to program within our community.  We do have other projects that require a form and tracks input values on a database.  We do have a team of new players that wish to help.  These are valuable to us as they are unindoctrinated and willing and eager with a fresh sight.

In it's present form, how do we determine if it is a functional app in our repository when we cloned it?  Or is it just a skeleton that needs to be compiled.  I imagine compiling like when you use .zip it compresses.  

Thanks again for any help thus far,

**Comment by JoahfromAlberta** [comment]
I plan on looking into programming soon, I will be studying standing up and alternating my focus from programming to other chores and duties.  In this manner can my body stay strong and more importantly my focus not shrunk consistently in my day.  I must embody my focus to be like a satellite with an orbit that extends towards my study and then outwards again to envelop healthy practices (such as organizing and cleaning) and then back again to my studies.  Only I can maintain my orbit through the belief that it is my life ring, and that it will save me.
